packages/manager/src/components/GenerateFirewallDialog/GenerateFirewallDialog.test.tsx

```diff
@@ -0,0 +1,110 @@
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+
+import { firewallFactory, firewallTemplateFactory } from 'src/factories';
+import { makeResourcePage } from 'src/mocks/serverHandlers';
+import { HttpResponse, http, server } from 'src/mocks/testServer';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { GenerateFirewallDialog } from './GenerateFirewallDialog';
+
+describe('GenerateFirewallButton', () => {
+  it('Can successfully generate a firewall', async () => {
+    const firewalls = firewallFactory.buildList(2);
+    const firewallTemplates = firewallTemplateFactory.buildList(2);
+    const createFirewallCallback = vi.fn();
+    const onClose = vi.fn();
+    const onFirewallGenerated = vi.fn();
+
+    server.use(
+      http.get('*/v4beta/networking/firewalls', () => {
+        return HttpResponse.json(makeResourcePage(firewalls));
+      }),
+      http.get('*/v4beta/networking/firewalls/templates', () => {
+        return HttpResponse.json(makeResourcePage(firewallTemplates));
+      }),
+      http.post('*/v4beta/networking/firewalls', async ({ request }) => {
+        const body = await request.json();
+        const payload = body as any;
+        const newFirewall = firewallFactory.build({
+          label: payload.label ?? 'mock-firewall',
+        });
+        createFirewallCallback(newFirewall);
+        return HttpResponse.json(newFirewall);
+      })
+    );
+
+    const { findByText, getByText } = renderWithTheme(
+      <GenerateFirewallDialog
+        onClose={onClose}
+        onFirewallGenerated={onFirewallGenerated}
+        open
+      />
+    );
+
+    getByText('Generate an Akamai Compliant Firewall');
+
+    act(() => {
+      userEvent.click(getByText('Generate Firewall Now'));
+    });
```

Comment:
I think `userEvent` should be awaited and I don't think we need the `act`

```suggestion
    await userEvent.click(getByText('Generate Firewall Now'));
```